packages/opentelemetry-exporter-collector/src/platform/node/CollectorExporterNodeBase.ts

```diff
@@ -0,0 +1,232 @@
+    this.credentials = config.credentials || grpc.credentials.createInsecure();
+    this.metadata = config.metadata;
+    this.headers =
+      parseHeaders(config.headers, this.logger) || this.DEFAULT_HEADERS;
+  }
+
+  onInit(config: CollectorExporterConfigNode): void {
+    this._isShutdown = false;
+    if (config.protocolNode === CollectorProtocolNode.HTTP_JSON) {
+      this.initWithJson();
+    } else {
+      this.initWithGrpc();
+    }
+  }
+
+  send(
+    spans: ExportItem[],
+    onSuccess: () => void,
+    onError: (error: collectorTypes.CollectorExporterError) => void
+  ): void {
+    if (this._isShutdown) {
+      this.logger.debug('Shutdown already started. Cannot send spans');
+      return;
+    }
+    if (this._protocol === CollectorProtocolNode.HTTP_JSON) {
+      this.sendWithJson(spans, onSuccess, onError);
+    } else {
+      this.sendWithGrpc(spans, onSuccess, onError);
+    }
+  }
+
+  onShutdown(): void {
+    this._isShutdown = true;
+    if (this.serviceClient) {
+      this.serviceClient.close();
+    }
+  }
+
+  getDefaultServiceName(config: CollectorExporterConfigNode): string {
+    return config.serviceName || DEFAULT_SERVICE_NAME;
+  }
+
+  protected initWithJson(): void {
+    // Nothing to be done for JSON yet
+  }
+
+  /**
+   * Initialize
+   */
+  protected initWithGrpc(): void {
+    const serverAddress = removeProtocol(this.url);
+    const includeDirs = [path.resolve(__dirname, 'protos')];
+
+    protoLoader
+      .load(this.getServiceProtoPath(), {
+        keepCase: false,
+        longs: String,
+        enums: String,
+        defaults: true,
+        oneofs: true,
+        includeDirs,
+      })
+      .then(packageDefinition => {
+        const packageObject: any = grpc.loadPackageDefinition(
+          packageDefinition
+        );
+        this.serviceClient = this.getServiceClient(
+          packageObject,
+          serverAddress
+        );
+        if (this.grpcQueue.length > 0) {
+          const queue = this.grpcQueue.splice(0);
+          queue.forEach((item: GRPCQueueItem<ExportItem>) => {
+            this.send(item.objects, item.onSuccess, item.onError);
+          });
+        }
+      });
+  }
+
+  protected sendWithGrpc(
+    objects: ExportItem[],
+    onSuccess: () => void,
+    onError: (error: collectorTypes.CollectorExporterError) => void
+  ): void {
+    if (this.serviceClient) {
+      const serviceRequest = this.convert(objects);
+
+      this.serviceClient.export(
+        serviceRequest,
+        this.metadata,
+        (err: collectorTypes.ExportServiceError) => {
+          if (err) {
+            this.logger.error('Service request', serviceRequest);
+            onError(err);
+          } else {
+            this.logger.debug('spans sent');
+            onSuccess();
+          }
+        }
+      );
+    } else {
+      this.grpcQueue.push({
+        objects,
+        onSuccess,
+        onError,
+      });
+    }
+  }
+
+  protected sendWithJson(
```

Comment:
why did you remove `utilWithJson.ts` and `utilWithGrpc.ts` and move the function here ?
I think it would be better to leave them there and just pass collector instance as it was so you get access to all what you need.